bin/propolis-server/src/lib/migrate/source.rs

```diff
@@ -137,17 +147,47 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> SourceProtocol<T> {
         self.read_ok().await
     }
 
-    async fn ram_push(&mut self) -> Result<(), MigrateError> {
-        self.update_state(MigrationState::RamPush).await;
+    async fn ram_push(
+        &mut self,
+        phase: &MigratePhase,
+    ) -> Result<(), MigrateError> {
+        // Only publish the "RAM push" phase before pausing (so that the
```

Comment:
I wanted to have avoid having the publicly-visible migration phase go from `RamPush` to `Pause` back to `RamPush` because that looked to me like the migration was going backwards. It turns out that the API already defines a separate `RamPushDirty` state for this, so we can just use that; fixed in 7d08ab7.